### PR TITLE
Static library bundles

### PR DESCRIPTION
**Pull request by edenhill**
Build static library bundles containing all external dependencies of librdkafka (or as many as possible).
This is to be used with confluent-kafka-go client ([demo here](https://github.com/confluentinc/confluent-kafka-go-example)).

Related Go PR:
https://github.com/confluentinc/confluent-kafka-go/pull/355

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -2,32 +2,52 @@ language: c
...
    os: osx
    compiler: gcc
+   env: LINKAGE=std
    before_script:
     - ./configure --install-deps --disable-lz4-ext --prefix="$PWD/dest"
  - name: "OSX clang: +static"
    os: osx
+   # Use an older image to and disable syslog for broader compatibility
+   # with old and new osx versions.
+   osx_image: xcode9.2
```
Comment (rnpridgeon) [inline .travis.yml]: Is this still necessary with the `--disable-syslog` flag?

Comment (edenhill) [reply .travis.yml]: Yes, older toolchains complained of .a files being generated by newer toolchains.

File: .travis.yml [hunk 2]
```diff
@@ -2,32 +2,52 @@ language: c
 dist: trusty
 sudo: false
 cache: ccache
+
 env:
-- ARCH=x64
+ global:
```
Comment (mhowlett) [inline .travis.yml]: inconsistent spacing/tab size of 1.

**Comment by mikekamornikov** [comment]
@edenhill What is the status of this one? Should we expect it (and https://github.com/confluentinc/confluent-kafka-go/pull/355) to me merged anytime soon? The thing should simplify a lot `bazel` builds. Currently I have to build `librdkafka` separately via `rules_foreign_cc` and do some magic to tie it to `confluent-kafka-go`. I can't cross compile the result this way (`bazel` complains about missing toolchain). And for me it means a switch to some pure go client.

**Pull request merged**